subsys/dfu/boot/mcuboot.c

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2016-2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <errno.h>
+#include <string.h>
+#include <flash.h>
+#include <zephyr.h>
+#include <init.h>
+
+#include <misc/__assert.h>
+#include "boot/mcuboot.h"
+
+/*
+ * Helpers for image trailer, as defined by mcuboot.
+ * Image trailer consists of sequence of fields:
+ *   u8_t copy_done
+ *   u8_t padding_1[BOOT_MAX_ALIGN - 1]
+ *   u8_t image_ok
+ *   u8_t padding_2[BOOT_MAX_ALIGN - 1]
+ *   u8_t magic[16]
+ */
+
+/*
+ * Strict defines: Defines in block below must be equal to coresponding
+ * mcuboot defines
+ */
+#define BOOT_MAX_ALIGN 8
+#define BOOT_MAGIC_SZ  16
+#define BOOT_FLAG_SET 0x01
+#define BOOT_FLAG_UNSET 0xff
+/* end_of Strict defines */
+
+#define BOOT_MAGIC_GOOD  1
+#define BOOT_MAGIC_BAD   2
+#define BOOT_MAGIC_UNSET 3
+
+#define BOOT_FLAG_IMAGE_OK 0
+#define BOOT_FLAG_COPY_DONE 1
+
+#define FLASH_MIN_WRITE_SIZE FLASH_ALIGN
+#define FLASH_BANK0_OFFSET FLASH_AREA_IMAGE_0_OFFSET
+#define FLASH_BANK_SIZE FLASH_AREA_IMAGE_0_SIZE
```

Comment:
ok, I think it is worth adding a comment, not all architectures and boards in Zephyr have DTS support. It will help understand where this is coming from.